src/ConnectedMode.UnitTests/ServerSentEvents/SSESessionManagerTests.cs

```diff
@@ -29,48 +30,85 @@ namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.ServerSentEvents;
 [TestClass]
 public class SSESessionManagerTests
 {
+    private const string DefaultProjectKey = "myproj";
+
     [TestMethod]
     public void MefCtor_CheckIsExported()
     {
+        var activeSolutionBoundTrackerMock = new Mock<IActiveSolutionBoundTracker>();
+        activeSolutionBoundTrackerMock.SetupGet(tracker => tracker.CurrentConfiguration)
+            .Returns(BindingConfiguration.Standalone);
+
         MefTestHelpers.CheckTypeCanBeImported<SSESessionManager, ISSESessionManager>(
-            MefTestHelpers.CreateExport<IActiveSolutionBoundTracker>(),
+            MefTestHelpers.CreateExport<IActiveSolutionBoundTracker>(activeSolutionBoundTrackerMock.Object),
             MefTestHelpers.CreateExport<ISSESessionFactory>());
     }
 
     [TestMethod]
     public void Ctor_SubscribesToBindingChangedEvent()
     {
-        var solutionTrackerMock = new Mock<IActiveSolutionBoundTracker>();
+        var testScope = new TestScope();
 
-        var _ = new SSESessionManager(solutionTrackerMock.Object, Mock.Of<ISSESessionFactory>());
+        var _ = testScope.CreateTestSubject();
 
-        solutionTrackerMock.VerifyAdd(
-            solutionTracker => solutionTracker.SolutionBindingChanged +=
+        testScope.ActiveSolutionBoundTrackerMock.VerifyAdd(
+            tracker => tracker.SolutionBindingChanged +=
                 It.IsAny<EventHandler<ActiveSolutionBindingEventArgs>>(), Times.Once);
     }
 
+    [TestMethod]
+    public void Ctor_ForceConnectsAfterSubscriptionToBindingChangedEvent()
+    {
+        var isSubscribed = false;
+        var testScope = new TestScope(TestScope.CreateConnectedModeBindingConfiguration(DefaultProjectKey));
+        testScope.ActiveSolutionBoundTrackerMock
+            .SetupAdd(solutionTracker => solutionTracker.SolutionBindingChanged += It.IsAny<EventHandler<ActiveSolutionBindingEventArgs>>()).
+            Callback(() => 
+            {
+                isSubscribed.Should().BeFalse();
+                isSubscribed = true;
+            });
+        testScope.SetUpSSEFactoryToReturnNoOpSSESession(DefaultProjectKey, () => isSubscribed.Should().BeTrue());
+
+        var _ = testScope.CreateTestSubject();
+
+        testScope.ActiveSolutionBoundTrackerMock.VerifyAdd(tracker => tracker.SolutionBindingChanged += It.IsAny<EventHandler<ActiveSolutionBindingEventArgs>>(), Times.Once);
+        testScope.SSESessionFactoryMock.Verify(factory => factory.Create(DefaultProjectKey), Times.Once);
+    }
+
+    [TestMethod]
+    public void Ctor_WhenInConnectedModeOnCreation_CreatesSession()
```

Comment:
What about the standalone test? with explicit verification that session factory was not called.